aten/src/ATen/native/cudnn/LossCTC.cpp

```diff
@@ -118,11 +123,42 @@ bool _use_cudnn_ctc_loss_tensor(
     const Tensor& input_lengths,
     const Tensor& target_lengths,
     int64_t BLANK) {
-  Tensor ilc = input_lengths.to(Device(at::kCPU), at::kLong).contiguous();
-  Tensor tlc = target_lengths.to(Device(at::kCPU), at::kLong).contiguous();
-  IntArrayRef il(ilc.data_ptr<int64_t>(), ilc.numel());
-  IntArrayRef tl(tlc.data_ptr<int64_t>(), tlc.numel());
-  return at::_use_cudnn_ctc_loss(log_probs, targets, il, tl, BLANK);
+  auto& ctx = at::globalContext();
+
+  bool use_cudnn = ctx.userEnabledCuDNN() && (BLANK == 0) &&
```

Comment:
I wonder if you need a v9 check here somewhere as well as fallback for ROCm (cc: @jeffdaily )

Reply:
v8 supports this: https://docs.nvidia.com/deeplearning/cudnn/latest/api/cudnn-adv-library.html#cudnnctcloss-v8 , but via the v8 descriptor call rather than the v9 one

Looks like CTCLoss has been skipped on ROCM thus far: `@skipCUDAIfRocm(msg="skipped Cudnn test on ROCm")`